File: rock_paper_scissors.rb
# Inspired by: https://github.com/feministy/codecademy-practice/blob/master/rock-paper-scissors.js
# And: https://github.com/feministy/Basic-practice/blob/master/exercises_dbc_socrates_prep/die_class_2_symbols.rb

class RockPaperScissors

  attr_accessor :user 

  def initialize
    @computer = computer
    @user = user
    @result = nil
  end

  def computer
    roll = rand(3)

    if roll == 0
      @computer = "rock"
    elsif roll == 1
      @computer = "scissors"
    else
      @computer = "paper"
    end 
    
    @computer  
  end

  def play 
    puts "Do you choose rock, paper, or scissors?"
    @user = gets.chomp
    
    victory = "You win! The computer picked #{@computer}, and you picked #{@user}, which is superior."
    defeat = "You lose! Womp Womp. The computer picked #{@computer}, and you picked #{@user}, which is inferior."

    if @user == @computer
      @result = "Ladies and gentlemen, we have a tie. The computer picked #{user}, too!"
    elsif @user == "rock"
      if @computer == "paper"
        @result = defeat
      else
        @result = victory
      end
    elsif @user == "paper"
      if @computer == "rock"
        @result = victory
      else
        @result = defeat
      end
    elsif @user == "scissors"
      if @computer == "rock"
        @result = defeat
      else
        @result = victory 
      end
    else
      @result = "You didn't pick rock, paper, or scissors. Silly goose!"
    end 

    puts @result if @result != nil
      
  end

end

new_game = RockPaperScissors.new
new_game.play

File: exercises_codecademy/rock-paper-scissors.js
var userChoice = prompt("Do you choose rock, paper or scissors?");
var computerChoice = Math.random();
if (computerChoice <0.34){
    computerChoice = "rock";
}
else if(computerChoice <=0.67){
  computerChoice = "paper";
}
else{
	computerChoice = "scissors";
}

var compare = function(choice1,choice2) {

if (choice1 === choice2) {
	console.log("The result is a tie!");
}

if (choice1 === "rock") {
	if (choice2 === "scissors"){
		console.log("rock wins");	
	}
else {
	console.log("paper wins");
}}

if (choice1 === "paper") {
	if (choice2 === "rock") {
		console.log("paper wins");}
else {
	console.log("scissors wins");
}}

if (choice1 === "scissors") {
	if (choice2 === "rock") {
		console.log("rock wins");
	}
else {
	console.log("scissors wins");
}}};
